### PR TITLE
feat(bigquery): Update to SDK V4

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

File: plugins/destination/bigquery/client/deletestale.go [hunk 1]
```diff
@@ -3,12 +3,11 @@ package client
...
-	"time"
 
-	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v4/message"
 )
 
 // DeleteStale is not currently implemented for BigQuery, as it only supports "append" write mode.
-func (*Client) DeleteStale(ctx context.Context, tables schema.Tables, sourceName string, syncTime time.Time) error {
+func (*Client) DeleteStale(context.Context, []*message.WriteDeleteStale) error {
 	return errors.New("DeleteStale not implemented")
```
Comment (disq) [inline plugins/destination/bigquery/client/deletestale.go:12]:
```suggestion
	return plugin.ErrNotImplemented
```

**Review by disq — DISMISSED** [review]
`if err := c.writer.Close(ctx); err != nil { return err }` is missing from `Client.Close()`

File: plugins/destination/bigquery/client/client.go [hunk 2]
```diff
@@ -2,43 +2,43 @@ package client
...
+	if err := json.Unmarshal(specBytes, &c.spec); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal BigQuery spec: %w", err)
 	}
-	spec.SetDefaults()
-	if err := spec.Validate(); err != nil {
+	c.spec.SetDefaults()
+	if err := c.spec.Validate(); err != nil {
+		return nil, err
+	}
+	c.writer, err = batchwriter.New(c, batchwriter.WithLogger(logger), batchwriter.WithBatchSize(c.spec.BatchSize), batchwriter.WithBatchSizeBytes(c.spec.BatchSizeBytes))
```
Comment (disq) [inline plugins/destination/bigquery/client/client.go:38]: missing batch timeout option

File: plugins/destination/bigquery/client/deletestale.go [hunk 3]
```diff
@@ -2,13 +2,12 @@ package client
...
 
-	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v4/message"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
 )
 
 // DeleteStale is not currently implemented for BigQuery, as it only supports "append" write mode.
-func (*Client) DeleteStale(ctx context.Context, tables schema.Tables, sourceName string, syncTime time.Time) error {
-	return errors.New("DeleteStale not implemented")
+func (*Client) DeleteStale(context.Context, message.WriteDeleteStales) error {
```
Comment (disq) [inline plugins/destination/bigquery/client/deletestale.go:11]: could just embed `batchwriter.UnimplementedDeleteStale` to `Client` and remove this file

File: plugins/destination/bigquery/client/client.go [hunk 4]
```diff
@@ -2,43 +2,43 @@ package client
...
-	logger     zerolog.Logger
-	spec       specs.Destination
-	metrics    destination.Metrics
-	pluginSpec Spec
-	client     *bigquery.Client
+	plugin.UnimplementedSource
+	logger zerolog.Logger
+	spec   Spec
+	client *bigquery.Client
+	writer *batchwriter.BatchWriter
```
Comment (disq) [inline plugins/destination/bigquery/client/client.go:20]: writer is still not closed in Close()

File: plugins/destination/bigquery/client/client.go (hunk 2 above)
Comment (hermanschaaf) [reply plugins/destination/bigquery/client/client.go:38]: Good catch

File: plugins/destination/bigquery/client/deletestale.go (hunk 3 above)
Comment (hermanschaaf) [reply plugins/destination/bigquery/client/deletestale.go]: updated

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**